scripts/build.ps1

```diff
@@ -219,15 +219,29 @@ function Publish-Package
     Write-Log "Package: Publish testhost.x86\testhost.x86.csproj"
     Publish-PackageInternal $testHostx86Project $TPB_TargetFramework $testhostFullPackageDir
 
-    # Copy over the Full CLR built testhost package assemblies to the $fullCLRPackageDir
-    Copy-Item $testhostFullPackageDir\* $fullCLRPackageDir -Force
-
-    # Copy over the Full CLR built testhost package assemblies to the Core CLR package folder.
+    # Binding redirection for Newtonsoft.Json.dll has been removed from testhost config file to resolve bug: https://github.com/Microsoft/vstest/issues/391 
+    # If a user's project is taking a dependency on Newtonsoft.Json with version 6.0.5 and in test code he is using Newtonsoft API's inside an appdomain. Appdomain uses config file of the parent process
+    # if not provided any, so here it will use testhost config file which has binding redirect <bindingRedirect oldVersion="0.0.0.0-9.0.0.0" newVersion="9.0.0.0" />. Since the user's projects is dependent on Newtonsoft.Json
+    # of version 6.0.5 their tests break because of the redirection.
+    # In the TestPlatform repo: Microsoft.TestPlatform.CommunicationUtilities depends on Newtonsoft.Json(8.0.3) and Microsoft.TestPlatform.CrossPlatEngine depends on Microsoft.Extensions.Dependencymodel which
+    # depends on Newtonsoft.Json(9.0.1). So testhost ends up having Newtonsoft.Json.dll of version 9.0.1 in their publish package.
+    # Since testhost uses CommunicationUtilities (and does not use that part of CrossPlatEngine which uses Newtonsoft.Json of version 9.0.1) so it needs Newtonsoft.Json.dll of version 8.0.3 along with it.
+    # Hence, copying Newtonsoft.Json.dll of version 8.0.3 to test $testhostFullPackageDir. This is however very much dependent on the CrossPlatEngine code running in testhost to not call into APIs specific to 9.0.1.
```

Comment:
We're copying the JSON.net since CommunicationUtilities requires it. Suggest adding this root cause to the actual bug and refer it here for history.

After refactoring of test host providers, CrossPlatEngine no longer takes a dependency on JSON.net.